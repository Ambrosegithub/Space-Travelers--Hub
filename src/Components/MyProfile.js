import React from 'react';
import Container from 'react-bootstrap/esm/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const myMissionProfile = useSelector((state) => [
    {
      MissionTitle: 'My mission',
      data: state.missions.filter((mission) => mission.reserved),
    },
  ]);
  return (
    <div className="content-div">
      <h1>My Missions</h1>
      {myMissionProfile.map(({ id, MissionTitle, data }) => (
        <Container key={id} className="d-grid">
          <Row className="missionprofile">
            <Col sm={6} className="title">{MissionTitle}</Col>
            <Col sm={6} className="no-mission">
              {!data.length ? (
                <p className="para">No Missions</p>
              ) : (
                data.map((item) => <p key={item.id} className="para1">{item.name}</p>)
              )}
            </Col>
          </Row>
        </Container>
      ))}
    </div>
  );
};

export default MyProfile;
