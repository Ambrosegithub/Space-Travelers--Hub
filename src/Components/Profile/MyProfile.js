import React from 'react';
import Container from 'react-bootstrap/esm/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { useSelector } from 'react-redux';
import RocketProfile from './Rockets/Rocket';

const MyProfile = () => {
  const myMissionProfile = useSelector((state) => [
    {
      MissionTitle: 'Missons',
      data: state.missions.filter((mission) => mission.reserved),
    },
  ]);
  return (
    <div>
      {myMissionProfile.map(({ id, MissionTitle, data }) => (
        <Container key={id}>
          <Row className=" col-3">
            <Col className="fw-bold" sm={5}>
              {MissionTitle}
            </Col>
            <div className="col-5">
              <Col sm={5}>
                {!data.length ? (
                  <p>No Missions</p>
                ) : (
                  // eslint-disable-next-line jsx-a11y/label-has-associated-control
                  data.map((item) => <label key={item.id}>{item.name}</label>)
                )}
              </Col>
            </div>
          </Row>
          <Row>
            <div className="profile-rockets col-5 ">
              <RocketProfile />
            </div>
          </Row>
        </Container>
      ))}
    </div>
  );
};

export default MyProfile;
