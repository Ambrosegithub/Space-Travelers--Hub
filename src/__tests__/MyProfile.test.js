import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MyProfile from '../Components/Profile/MyProfile';
import store from '../redux/configureStore';

describe('Test of Profile Components', () => {
  it('Renders MyProfile component correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
