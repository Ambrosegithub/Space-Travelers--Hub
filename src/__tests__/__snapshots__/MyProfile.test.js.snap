// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Profile Components Profile Components Renders correctly 1`] = `
<div
  className="container-fluid container"
>
  <table
    className="col-xl-12 col-lg-12 col-md-12 col-sm-12 table"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`Test of Profile Components Renders MyProfile component correctly 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className=" col-3 row"
    >
      <div
        className="fw-bold col-sm-5"
      >
        Missons
      </div>
      <div
        className="col-5"
      >
        <div
          className="col-sm-5"
        >
          <p>
            No Missions
          </p>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="profile-rockets col-5 "
      >
        <table
          className="zebra"
        >
          <thead>
            <tr>
              <th
                className="table-title"
              >
                Rockets
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                className="table-title"
              >
                No missions available ..
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
